scripts/area_plot.py

```diff
@@ -0,0 +1,613 @@
+        included.
+    group_thresh_fine, group_thresh_rough: int, [1, 6]
+        The maximum number of taxonmic groups to display for the respective 
+        level. If `group_thresh_rough` > 6, then it will be replaced with 
+        6 because this is the maximum number of avaliable color groups.
+
+    Returns
+    -------
+    Figure
+        A 8" x 4" matplotlib figure with the area plot and legend.
+
+    Also See
+    --------
+    single_area_plot
+
+    """
+
+    # Parses the taxonomy and collapses the table
+    taxa = extract_label_array(table, tax_col, tax_delim)
+    if samples is None:
+    	samples = list(table.drop(columns=[tax_col]).columns)
+
+    # Gets the appropriate taxonomic level information to go forward
+    collapsed = level_taxonomy(table, taxa, samples, 
+                               level=np.array([fine_level]), 
+                               consider_nan=multilevel_table)
+    samples = collapsed.columns
+
+    # Gets the top taxonomic levels
+    upper_, lower_, = profile_joint_levels(collapsed, rough_level, fine_level, 
+    								       samples=samples,
+                                           lo_thresh=abund_thresh_rough, 
+                                           lo_count=min(5, group_thresh_rough),
+                                           hi_thresh=abund_thresh_fine,
+                                           hi_count=group_thresh_fine,
+                                           )
+    # Gets the colormap 
+    cmap = define_join_cmap(upper_)
+    upper_.index = upper_.index.droplevel('rough')
+    lower_.index = lower_.index.droplevel('rough')
+
+    # Plots the data
+    fig_ = plot_area(upper_.astype(float), lower_.astype(float), cmap)
+
+    return fig_
+
+
+# Sets up the main arguments for argparse.
+parser_one = argparse.ArgumentParser(
+    description=('A set of functions to generate diagnostic stacked area '
+                 'plots from metagenomic outputs.'),
+    prog=('area_plotter'),
+    )
+parser_one.add_argument('-t', '--table', 
+                    help=('The abundance table as a tsv classic biom '
+                          '(features as rows, samples as columns) containing'
+                          'absloute or relative abundance for the samples.'),
+                    required=True,
+                    )
+parser_one.add_argument('-f', '--figure',
+                    help=('The location for the final figure'),
+                    required=True,
+                    )
+parser_one.add_argument('-s', '--samples', 
+                    help=('A text file with the list of samples to be include' 
+                         ' (one per line). If no list is provided, then data '
+                         'from all columns in the table (except the one '
+                         'specifying taxonomy) will be used.'),
+                    )
+parser_one.add_argument('--mode', choices=(["metaphlan", "kracken", "marker"]),
```

Comment:
I love that you added `--mode`. An additional improvement I would like to see is that using `--mode` will only change the settings of all the different parameters. For example, make a small dictionary with the different settings for the different modes:

```
modes = {
    "metaphlan": {
        "tax_delim": "|", 
        "multi_level": True
        "tax_col": "clade_name"
        "table_drop": "NCBI_tax_id"
        "skip_rows": 1
    },
    "kraken": {
        ... etc.
    },
    "marker": {
        ... etc.
    },
    ... etc
}
```

Set the default of `args.mode` to an empty string, then test if the string is empty and then use the `dict.update` method on the `Namespace.__dict__` object you get from `ArgumentParser.parse()` to push the mode-specific settings from the mode dictionary (if the user selects a mode, that is). Otherwise all the defaults are already set to something meaningful (e.g. kraken, or metaphlan-style). 

This implementation makes it very easy for users to run the tool on a format that is slightly different from what we anticipated, but still flexible and easy enough to implement for us.

Reply:
By the way, all references to Kraken-style should be changed to `kraken` instead of `kracken`.